vertico-prescient.el

```diff
@@ -0,0 +1,190 @@
+  "Previous binding of `M-s' in `vertico-map'.")
+
+(defun vertico-prescient--remember ()
+  "Advice for remembering candidates in Vertico."
+  (when (>= vertico--index 0)
+    (prescient-remember
+     (substring-no-properties
+      (nth vertico--index vertico--candidates)))))
+
+(defvar-local vertico-prescient--local-settings nil
+  "Whether this buffer has local settings due to `vertico-prescient-mode'.")
+
+(defun vertico-prescient--apply-completion-settings ()
+  "Enable `prescient-completion-local-mode' after `vertico--setup'."
+  (unless prescient-completion-local-mode
+    (prescient-completion-local-mode 1)
+    (setq vertico-prescient--local-settings t)))
+
+;;;###autoload
+(define-minor-mode vertico-prescient-mode
+  "Minor mode to use prescient.el in Vertico menus.
+
+This mode will:
+- if `vertico-prescient-override-sorting' is non-nil,
+  configure `vertico-sort-override-function' and set
+ `vertico-prescient-enable-filtering' to t
+
+- if `vertico-prescient-enable-filtering' is non-nil,
+  configure `vertico-sort-function'
+
+- if `vertico-prescient-enable-filtering' is non-nil:
+  - enable `prescient-completion-local-mode' in the Vertico buffer
+  - bind `prescient-toggle-map' to `M-s' in `vertico-map'
+
+- advise `vertico-insert' to remember candidates"
+  :global t
+  :group 'prescient
+  (if vertico-prescient-mode
+      ;; Turn on the mode.
+      (progn
+        ;; Prevent messing up variables if we explicitly enable the
+        ;; mode when it's already on.
+        (vertico-prescient-mode -1)
+        (setq vertico-prescient-mode t)
+
+        (when vertico-prescient-override-sorting
+          (setq vertico-prescient-enable-sorting t)
+          (cl-shiftf vertico-prescient--old-sort-override-function
+                     vertico-sort-override-function
+                     #'prescient-completion-sort))
+
+        (when vertico-prescient-enable-sorting
+          (cl-shiftf vertico-prescient--old-sort-function
+                     vertico-sort-function
+                     #'prescient-completion-sort))
+
+        (when vertico-prescient-enable-filtering
+          ;; Configure completion settings.
+          (advice-add 'vertico--setup
+                      :after #'vertico-prescient--apply-completion-settings)
+
+          ;; Bind toggling commands.
+          (setq vertico-prescient--old-toggle-binding
+                (lookup-key vertico-map (kbd "M-s")))
+          (define-key vertico-map (kbd "M-s") prescient-toggle-map)
+
+          ;; Make sure Vertico refreshes immediately.
+          (add-hook 'prescient--toggle-refresh-functions
+                    #'vertico-prescient--toggle-refresh))
+
+        ;; While sorting might not be enabled in Vertico, it might
+        ;; still be enabled in another UI, such as Company or Corfu.
+        ;; Therefore, we still want to remember candidates.
+        (advice-add 'vertico-insert :after #'vertico-prescient--remember))
```

Comment:
I think it would feel more consistent to only install the advice if sorting is enabled.

Reply:
Maybe, but because the candidate selection should affect sorting in other UIs using `prescient.el`, I think that the advice should always be added.

Reply:
I argue that this would be quite unexpected. If Vertico/Corfu don't participate in Prescient sorting, why should they notify Prescient about their selected candidates? Furthermore if you disable Prescient sorting you may not even want to use the sorting functionality at all (only the filtering). Then you wouldn't want the remembering functionality to be used.

Anyway, I don't feel strongly about this issue. It is okay if you leave it as is.

Reply:
Leaving as is.